RxSwift/Schedulers/VirtualTimeScheduler.swift

```diff
@@ -106,7 +108,10 @@ open class VirtualTimeScheduler<Converter: VirtualTimeConverterType>
      - returns: The disposable object used to cancel the scheduled action (best effort).
      */
     public func scheduleAbsoluteVirtual<StateType>(_ state: StateType, time: VirtualTime, action: @escaping (StateType) -> Disposable) -> Disposable {
-        MainScheduler.ensureExecutingOnScheduler()
+        guard thread == nil || Thread.current == thread else {
+            rxFatalError("Executing on the wrong thread. Please ensure all work on the same thread.")
+        }
+        thread = Thread.current
```

Comment:
Is it safe to read and write `thread` without any locking?

Reply:
Isn't that what we always do on the main thread? Read/write without locking? Maybe I don't understand the question...

Reply:
Indeed, it should be completely fine as long scheduler's methods are actually called on one thread, but what if there would be two close calls from different threads, it's not impossible that both will see `thread == nil` and proceed, for example.

Reply:
Right! Good point. I will update to account for that.

Reply:
@danielt1263 sounds good!